front/src/App.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div id="app">
+    <a href="https://github.com/pniedzwiedzinski/get-lucky" class="GitHub">
+      <img src="../assets/GitHub.png" alt="GitHub">
+    </a>
+    <div class="container">
+      <h1>Szczęśliwy Numerek</h1>
+      <div class="loading">
+        <span class="dot dot1"></span>
+        <span class="dot dot2"></span>
+        <span class="dot dot3"></span>
+        <h2>{{number}}</h2>
+      </div>
+      <h4>{{error}}</h4>
+      <div class="curl">
+        <h4 style="color: #DE3131;">curl </h4>
+        <h4>https://get-lucky.netlify.com/.netlify/functions/get</h4>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  name: 'app',
+  el: '#loading',
+  data() {
+    return {
+      number: null,
+      error: null,
+    }
+  },
+  mounted () {
+    this.getLuckyNumber();
+  },
+  methods: {
+    getLuckyNumber() {
+      axios.get('https://get-lucky.netlify.com/.netlify/functions/get')
+      .then(response => {
+        this.number = response.data.data;
+        const dot = document.querySelectorAll(".dot");
+        dot.forEach(item => {
+          item.style.display = "none";
+        });
+      })
+      .catch(error => {
+        this.error = error;
+      })
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,700&display=swap');
+
+*,*::after,*::before {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+body {
+  font-family: 'Roboto', sans-serif;
+  color: #03303F;
+  background-color: #FCFCF1;
+}
+
+[v-cloak] > * {
+  display: none;
+}
+
+[v-cloak]::before {
+  content: '';
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  width: 1rem;
+  height: 1rem;
+  background-color: black;
+  z-index: 1;
+}
+
+.container {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  text-align: center;
+
+  h1 {
+    font-size: 3.5rem;
+    text-transform: uppercase;
+    margin-bottom: 1rem;
+  }
+
+  .loading {
+    h2 {
+      margin: 3rem 0 8rem 0;
+      font-size: 12rem;
+      font-weight: 700;
+    }
+    .dot{
+      height: 10px;
+      width: 10px;
+      border-radius: 50%;
+      background-color: #03303F;
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+    }
+    .dot1 {
+      left: 47%;
+      animation: loading 1s linear infinite;
+    }
+    .dot2 {
+      animation: loading 1s linear .25s infinite;
+    }
+    .dot3 {
+      left: 53%;
+      animation: loading 1s linear .5s infinite;
+    }
```

Comment:
Ja już używasz scss'a to czemu by nie uprościć tego?

Reply:
Nie rozumiem o co konkretnie chodzi co tutaj można uprościć?

Reply:
https://sass-lang.com/documentation/at-rules/mixin

Reply:
Zmienione